arch/eventloop_lws_http.c

```diff
@@ -0,0 +1,259 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*
+*   Copyright 2023 (c) Fraunhofer IOSB (Author: Noel Graf)
+*/
+
+/* The HTTP ConnectionManager is architecture agnostic. It selects an existing
+* TCP ConnectionManager and uses it for the underlying TCP connections. */
+
+/* Suppress the warning 'redefinition of typedef' in the libwebsockets library.
+ * Removed as soon as the patch is included in the main. */
+#pragma GCC diagnostic ignored "-Wpedantic"
+
+#include <open62541/plugin/eventloop.h>
+#include "open62541/plugin/log_stdout.h"
+
+#include "lws/ua_lws_adapter.h"
+
+struct HTTPConnectionManager;
+typedef struct HTTPConnectionManager HTTPConnectionManager;
+
+#define HTTP_PARAMETERSSIZE 5
+
+static const struct {
+    UA_QualifiedName name;
+    const UA_DataType *type;
+    UA_Boolean required;
+} HTTPSendParameters[HTTP_PARAMETERSSIZE] = {
+    {{0, UA_STRING_STATIC("url")}, &UA_TYPES[UA_TYPES_STRING], true},
```

Comment:
You mean address instead of url.
The url would include the port and path as well.
How would this be extended for https?
You can already add this as a comment.